File: fix.php
<?php
$showContent = file('resources/views/kepengurusan/anggota/proker/show.blade.php', FILE_IGNORE_NEW_LINES);
$progContent = file('resources/views/kepengurusan/anggota/proker/progress.blade.php', FILE_IGNORE_NEW_LINES);

$showStart = -1;
foreach ($showContent as $i => $line) {
    if (strpos($line, '<!-- Riwayat Progres Full -->') !== false) {
        $showStart = $i;
        break;
    }
}
$timelineBlock = array_slice($showContent, $showStart, 87);

$progDelStart = -1;
foreach ($progContent as $i => $line) {
    if (strpos($line, '<!-- Riwayat Jurnal -->') !== false) {
        $progDelStart = $i;
        break;
    }
}
$progDelEnd = -1;
for ($i = $progDelStart; $i < count($progContent); $i++) {
    if (strpos(trim($progContent[$i]), '<!-- Kolom Kanan: Info Ringkas Progres -->') !== false) {
        $progDelEnd = $i - 2;
        break;
    }
}

array_splice($progContent, $progDelStart, $progDelEnd - $progDelStart);

$insertPos = -1;
for ($i = count($progContent) - 1; $i >= 0; $i--) {
    if (trim($progContent[$i]) === '@endsection') {
        $insertPos = $i - 1; // Before @endsection, and the closing divs
        break;
    }
}
// Actually, let's insert before the closing '</div>' of the right column.
// The right column is the last div before @endsection.
for ($i = count($progContent) - 1; $i >= 0; $i--) {
    if (trim($progContent[$i]) === '<!-- Kolom Kanan: Info Ringkas Progres -->') {
        $insertPos = $i + 17; // This is a rough estimation, let's find the closing div of space-y-6
        break;
    }
}

// better way to find the insertion point:
// find '<!-- Kolom Kanan: Info Ringkas Progres -->'
$rightColStart = -1;
foreach ($progContent as $i => $line) {
    if (strpos(trim($line), '<!-- Kolom Kanan: Info Ringkas Progres -->') !== false) {
        $rightColStart = $i;
        break;
    }
}

$insertPos = $rightColStart;
for ($i = $rightColStart; $i < count($progContent); $i++) {
    if (trim($progContent[$i]) === '</div>' && trim($progContent[$i+1]) === '</div>' && trim($progContent[$i+2]) === '@endsection') {
        $insertPos = $i;
        break;
    }
}

array_splice($progContent, $insertPos, 0, $timelineBlock);

file_put_contents('resources/views/kepengurusan/anggota/proker/progress.blade.php', implode("\n", $progContent) . "\n");
echo "Done\n";

File: fix.py
import sys

def process():
    with open('resources/views/kepengurusan/anggota/proker/show.blade.php', 'r', encoding='utf-8') as f:
        show_content = f.read().splitlines()
    
    with open('resources/views/kepengurusan/anggota/proker/progress.blade.php', 'r', encoding='utf-8') as f:
        prog_content = f.read().splitlines()

    # Find Riwayat Progres Full in show.blade.php
    show_start = -1
    for i, line in enumerate(show_content):
        if '<!-- Riwayat Progres Full -->' in line:
            show_start = i
            break
            
    # extract timeline block (87 lines based on our previous file view)
    timeline_block = show_content[show_start:show_start+87]

    # find where to delete in progress.blade.php
    prog_del_start = -1
    for i, line in enumerate(prog_content):
        if '<!-- Riwayat Jurnal -->' in line:
            prog_del_start = i
            break
            
    prog_del_end = -1
    for i in range(prog_del_start, len(prog_content)):
        if prog_content[i].strip() == '<!-- Kolom Kanan: Info Ringkas Progres -->':
            prog_del_end = i - 2
            break
            
    # delete the block
    new_prog = prog_content[:prog_del_start] + prog_content[prog_del_end:]
    
    # Actually, we need to insert it at the end of Kolom Kanan
    insert_pos = -1
    for i in range(len(new_prog)-1, -1, -1):
        if new_prog[i].strip() == '@endsection':
            insert_pos = i - 2
            break
            
    final_prog = new_prog[:insert_pos] + [''] + timeline_block + new_prog[insert_pos:]
    
    with open('resources/views/kepengurusan/anggota/proker/progress.blade.php', 'w', encoding='utf-8') as f:
        f.write('\n'.join(final_prog) + '\n')
        
process()
